### ISSUE TITLE
Hosting Flask App in Subdirectory

### ISSUE THREAD

**Issue by srianbury** [MicrosoftDocs/visualstudio-docs#2901]
Thanks for the great docs! I was wondering if there is anything special needed to host a flask app on IIS in a subdirectory of a website?  In React you have to set homepage in the package.json file is there anything like that needed for flask?  When I right click the app in IIS, then click "browse" under "Manage Application" I get the message "Not Found
The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again".
Edit: I should also mention I am using HttpPlatform handler
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ed35685d-e409-d962-c450-7297a1f3c003
* Version Independent ID: 224fab0f-fd26-396f-cdd8-921533d2aeba
* Content: [Configure Python web apps for IIS - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/python/configure-web-apps-for-iis-windows?view=vs-2017#feedback)
* Content Source: [docs/python/configure-web-apps-for-iis-windows.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/python/configure-web-apps-for-iis-windows.md)
* Product: **visual-studio-windows**
* GitHub Login: @kraigb
* Microsoft Alias: **kraigb**

**Comment by kraigb** [comment]
I'm not sure (having not testing it), but you *might* be able to just add the subfolder to the path for WSGI_HANDLER, e.g. value="app_folder\flask_iis_example.app". If that doesn't work, try a colon instead of a \. (The latter might require the subfolder to be a Python module, meaning that you'd need an empty __init__.py in that folder.) Again, I haven't tried these, I'm just suggesting.

Alternately, you could place a script with an object in the root folder that returns the subfolder app object. In that case, the value would include () to make the function call rather than referring to the object directly.

A helpful reference: https://stackoverflow.com/questions/5072166/how-do-i-deploy-a-flask-application-in-iis

**Comment by srianbury** [comment]
I kind of figured it out.  I have a file called runserver.py that looks like this: 
```
from os import environ
from FlaskExample import app

if __name__ == '__main__':
    HOST = environ.get('SERVER_HOST', 'localhost')
    try:
        PORT = int(environ.get('SERVER_PORT', '5555'))
    except ValueError:
        PORT = 5555
    app.run(HOST, PORT)
```
I put the app in it's own website and set its http port to 5555 and it loaded externally and locally.  I think I just need to use `app.run()` in `runserver.py`, but I am still figuring it out.

**Comment by srianbury** [comment]
I was able to resolve this problem.  I had to do two things to do this.  

1. Serve the app with gevent or something other than flask (as it says in the [docs](http://flask.pocoo.org/docs/1.0/deploying/), my mistake).  
2. After this there was an issue accessing my app from the sub-directory path i.e. `mywebsite.com/app-name`. I was already using an ["application factory"](http://flask.pocoo.org/docs/1.0/patterns/appfactories/) to register my api endpoints and I just had to set the `url_prefix` parameter in the `register_blueprint` function to my sub directory name i.e. `app.register_blueprint(crud, url_prefix='/app-name')`.  I appreciate your help!
